### PR TITLE
Add getter for currently active tab

### LINKED ISSUE #15: Why are there no getters for tabs?
Hi,

as far as I can see, there are no getters for the current (or any other) tab in the interface, is that right?

Why not?

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Closes #15

### CODE REVIEW

**Comment by fin-ger** [comment]
According to the [rust naming conventions](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#getter/setter-methods-[rfc-344]) we have `active_tab()` and `set_active_tab()` both implemented on `TabView` and `TabPanel`. A `get_active_tab()` would therefore violate this naming convention. However, I think you mean an ambiguity in our API design where `active_tab()` is not providing the tab itself but the identifying key.

First of all, the choice behind this was that users can use a `NamedView` with the `Nameable` trait to access a tabs view (by simply wrapping a tab with a `NamedView`). We wanted to avoid yet-another-identifying mechanism in cursive (there also was the whole id vs. name discussion in cursive itself). That's also why we are returning a key for the current tab and not the view. Also, the key is used to reorder, remove, swap tabs, etc. It seemed reasonable to stick to the 'a tab is just a key' approach (remember a key can easily be the name of the tab by choosing a string as the tab's key like in our examples).

However, this adds some unexpected behaviour to our API as it is not straightforward to get the `View` of a tab. We wanted to avoid to force each tab to be wrapped with a `NamedView` and also let the user decide on the names. However, redesigning the API to match a key to the tab's name might be a better approach from a user's perspective. This would add some restrictions to the key data type, as it must fit into the type constraints of a `NamedView` (which I think was just a `String`). This means no `u8` or similar will be possible as tab identifiers. But I think this should have no performance impact on our `TabPanel`. There might be little performance penalties by forcing each tab to be wrapped in a `NamedView` but I think this would be ok.

@jwuensche What are your thoughts on this?

File: src/lib.rs [hunk 1]
```diff
@@ -101,6 +101,19 @@ impl<K: Hash + Eq + Clone + 'static> TabView<K> {
         self.current_id.as_ref()
     }
 
+    pub fn get_active_tab(&self) -> Option<&Box<dyn View>> {
```
Comment (gyscos) [inline src/lib.rs:104]: Do you need to expose the `Box` layer here? Returning `Option<&dyn View>` may fit most use-cases, is a bit simpler, and keeps the actual storage internal (so it'd be easy to move to `Rc` or other without breakage).

Comment (matthiasbeyer) [reply src/lib.rs:104]: Yep, sounds like a good idea. I'll wait on the discussion in #15 tho, maybe this PR gets superseded by another more general rework of the API. :small_airplane: :smile:

Comment (jwuensche) [reply src/lib.rs:104]: A good idea :D We should do this to allow more choices for the user of the library. Also, with the discussion at https://github.com/deinstapel/cursive-tabs/issues/15 we decided that this PR will be included :D
I'll add a few commits to update the PR to reflect the changes proposed here and then add the PR.

**Pull request merged**